### DISCUSSION TITLE
Error after upgrading to 1.1.15

### DISCUSSION THREAD

**Discussion by sysopdm** [Ylianst/MeshCentral#5495] [Issues]
After upgrading to version 1.1.15 I'm seeing the following in the error log:

-------- 11/1/2023, 5:53:19 PM ---- 1.1.15 --------

node:internal/modules/cjs/loader:1051
  throw err;
  ^

Error: Cannot find module 'mongodb'
Require stack:
- /opt/mc/node_modules/meshcentral/db.js
- /opt/mc/node_modules/meshcentral/meshcentral.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1048:15)
    at Module._load (node:internal/modules/cjs/loader:901:27)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at module.exports.CreateDB (/opt/mc/node_modules/meshcentral/db.js:812:9)
    at CreateMeshCentralServer.obj.StartEx (/opt/mc/node_modules/meshcentral/meshcentral.js:807:28)
    at CreateMeshCentralServer.obj.Start (/opt/mc/node_modules/meshcentral/meshcentral.js:438:66)
    at /opt/mc/node_modules/meshcentral/meshcentral.js:4103:24
    at /opt/mc/node_modules/meshcentral/meshcentral.js:3887:9
    at ChildProcess.exithandler (node:child_process:414:7) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/opt/mc/node_modules/meshcentral/db.js',
    '/opt/mc/node_modules/meshcentral/meshcentral.js'
  ]
}

-------- 11/1/2023, 5:53:19 PM ---- 1.1.15 --------

Node.js v20.9.0

Everything seems to be working at the moment, just concerned I might have an issue still.  Any ideas?

**Comment by si458** [comment]
It seems to think you are using mongodb, what's ur config.json like?

**Reply by sysopdm** [reply]
Hey Simon, I am using Monogodb.  Is that not supported anymore?

**Reply by si458** [reply]
Its always been supported and still is?
We havent changed anything relating to mongodb?
Can you share ur config.json at all please? (Hide secret stuff obvs)

**Reply by sysopdm** [reply]
Sure.

{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "Left-Out",
    "WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "port": 443,
    "MongoDb": "mongodb://127.0.0.1:27017",
    "_aliasPort": 443,
    "redirPort": 80,
    "_redirAliasPort": 80
  },
  "domains": {
    "": {
      "title": "Left-Out",
      "_title2": "Left-Out",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  },
  "letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "Left-Out",
    "names": "Left-Out",
    "rsaKeySize": 3072,
    "production": true
  }
}

**Comment by itpilot** [comment]
Guys please can you help me

**UBUNTU 20.04.6 LTS X86_64**

`root@meshcentral:~$ **node -v
v10.19.0**

root@meshcentral:~$ **/usr/bin/node -v
v10.19.0`**

i am having the same issue after **upgrading from version 1.1.13 to version 1.1.16**

**My config.json**
{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": Left-Out",
    "_TrustedCert": true,
    "WANonly": true,
    "WsCompression": true,
    "RelayDNS": "Left-Out",
    "AgentWsCompression": true,
    "MongoDb": "mongodb://127.0.0.1:27017/meshcentral",
    "MongoDbBulkOperations": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "port": 443,
    "RelayPort": 453,
    "aliasPort": 443,
    "plugins": { "enabled": true },
    "allowLoginToken": true,
    "allowFraming": true,
    "autoBackup": {
        "backupIntervalHours": 24,
        "keepLastDaysBackup": 3,
	    "zipPassword": "Left-Out",
            "GoogleDrive": {
            "folderName": "Left-Out",
            "maxFiles": 10
        }
      }, 
     "webrtc": true,
    "_TlsOffload": "127.0.0.1",
    "redirPort": 80,
    "redirAliasPort": 80
  },
  "domains": {
    "": {
      "SiteStyle": 2,
      "title": "Left-Out",
      "_title2": "Servername",
      "_minify": true,
      "GeoLocation": true,
      "_newAccounts": true,
      "_userNameIsEmail": true,
      "_usersessionidletimeout": 30,
      "WelcomeText": "Left-Out",
      "footer": "<a href='Left-Out</a>",
      "agentCustomization": {
        "_displayName": "Left-Out",
        "description": "Left-Out",
        "_companyName": "Left-Out",
        "image": "Left-Out",
	"installText": "Left-Out",  
	"altmessenging": {
      "name": "Jitsi",
      "url": "Left-Out"
	    }
      }
    }
  },
  "letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "Left-Out",
    "names": "Left-Out",
    "production": true
},
   "smtp": {
      "host": "Left-Out", 
      "port": Left-Out, 
      "from": "Left-Out", 
      "user": "Left-Out", 
      "pass": "Left-Out", 
      "tls": false
   }
}


**Where can i find the Log Files to see what the issue is?**

If i need to upgrade NODE how do i do that?

any help greatly appreciated

**Reply by si458** [reply]
for starters you are using an OLD version of node upgrade node if ur using ubuntu - https://github.com/nodesource/distributions#installation-instructions

**Reply by si458** [reply]
also mongo requires at least node 16!
```
"engines": {
    "node": ">=16.20.1"
}
```

**Comment by itpilot** [comment]
Thank you very much for the link and explanation

I have done as per your instructions

Please find the output via cli
***@***.***:~# *node -v*
v20.9.0
***@***.***:~# */usr/bin/node -v*
v20.9.0

*I did reboot the Meshcentral server However I KEEP GETTING THE FOLLOWING
ERROR:*

This site can’t be reached

*mesh.mywebsite.com <http://mesh.mywebsite.com>* refused to connect.


I have tried to see what is going on

***@***.***:/opt# *sudo systemctl status meshcentral.service*
● meshcentral.service - MeshCentral Server
     Loaded: loaded (/etc/systemd/system/meshcentral.service; enabled;
vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Tue
2023-11-21 04:51:59 UTC; 4s ago
    Process: 4279 ExecStart=/usr/bin/node
/opt/meshcentral/node_modules/meshcentral (code=exited, status=1/FAILURE)
   Main PID: 4279 (code=exited, status=1/FAILURE)
***@***.***:/opt#

I have restored from my backup

Any help greatly appreciated

**Reply by si458** [reply]
Meshcentral isn't starting up correctly,
Is the a mesherrors.log file inside meshcentral-data? Is the any data inside it?

**Comment by itpilot** [comment]
si458 Thank you for your support

Is the mesherrors.log file inside meshcentral-data? Is the any data inside it - **YES**
[mesherrors.txt](https://github.com/Ylianst/MeshCentral/files/13427027/mesherrors.txt)

I have restarted the Meschcentral Server

The NODE Upgrade has completed and i can Login with 2FO

The problem is that it does log me in then for a brief moment i can see my Meshcentral Web Page and Devices then it Automatically LOGS ME OUT  

Any ideas?

**Reply by si458** [reply]
mesherrors.txt shows ` } catch (ex) { console.log('Error loading plugin handler (' + e + ')'); }` is the lines its crashing on
so it things you have plugins enabled and a plugin is faulty/incompitable
can you disable plugins in your config.json and see what happens?
also can you share ur config.json (hide secrets)

EDIT: also the line should be erroring out ex not e! will submit a fix for this line

**Comment by itpilot** [comment]
si458 Thank you very much for your quick response, much appreciated

Yes it was a Plugin that was the issue

I just disabled the plugins option in the **config.json** file then did a reboot

All is functional and working

**Thank you very much for a Fantastic Application**
